plugins/authentication-middleware/src/providers/unverifiedJWTMiddlewareProvider.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AuthenticationMiddlewareProvider } from '.';
+import { JsonValue } from '@backstage/types';
+import { InputError, stringifyError } from '@backstage/errors';
+
+export const unverifiedJWTMiddlewareProvider: AuthenticationMiddlewareProvider =
```

Comment:
This is what the scaffolder is currently doing. Extracting the jwt token from the header, parsing it and extracting the ent and sub payload fields and setting them as the identity.